Panic on struct field pointing to zero value

When using `ValidateStruct` on a struct that has a field using a pointer to any field that contains a zero value, the validation panics.

```go
package main

import validation "github.com/go-ozzo/ozzo-validation"

type A struct {
	Name *string
}

func (a *A) Validate() error {
	return validation.ValidateStruct(a,
		validation.Field(&a.Name, validation.Length(1, 32)),
	)
}
func main() {
	var a A
	a.Validate()
}

```

```sh
$ go run main.go
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 1 [running]:
panic(0x117440, 0xc420206800)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
reflect.Value.Type(0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:1670 +0x224
github.com/go-ozzo/ozzo-validation.Validate(0x119ca0, 0x0, 0xc420193c90, 0x1, 0x1, 0x196, 0xc4202052d0)
```

--- a/struct_test.go
+++ b/struct_test.go
@@ -66,6 +66,7 @@ func TestValidateStruct(t *testing.T) {
 		{"t2.1", &m1, []*FieldRules{Field(&m1.A, &validateAbc{}), Field(&m1.B, &validateXyz{})}, ""},
 		{"t2.2", &m1, []*FieldRules{Field(&m1.A, &validateXyz{}), Field(&m1.B, &validateAbc{})}, "A: error xyz; B: error abc."},
 		{"t2.3", &m1, []*FieldRules{Field(&m1.A, &validateXyz{}), Field(&m1.c, &validateXyz{})}, "A: error xyz; c: error xyz."},
+		{"t3.4", &m1, []*FieldRules{Field(&m1.H, Length(0, 5))}, ""},
 		// non-struct pointer
 		{"t3.1", m1, []*FieldRules{}, StructPointerError.Error()},
 		{"t3.2", nil, []*FieldRules{}, StructPointerError.Error()},

--- a/validation.go
+++ b/validation.go
@@ -63,6 +63,11 @@ func Validate(value interface{}, rules ...Rule) error {
 	if rv.Kind() == reflect.Ptr || rv.Kind() == reflect.Interface {
 		rv = rv.Elem()
 	}
+
+	if !rv.IsValid() {
+		// Skip zero values after rule validations.
+		return nil
+	}
 	rt, rk := rv.Type(), rv.Kind()
 
 	if rk == reflect.Map && rt.Elem().Implements(validatableType) {

--- a/validation_test.go
+++ b/validation_test.go
@@ -5,6 +5,7 @@
 package validation
 
 import (
+	"encoding/json"
 	"errors"
 	"strings"
 	"testing"
@@ -108,6 +109,7 @@ type Model1 struct {
 	E String123
 	F *String123
 	G string `json:"g"`
+	H *json.RawMessage
 }
 
 type String123 string

